sites/all/themes/orwell/template.php

```diff
@@ -33,6 +33,33 @@ function orwell_form_search_block_form_alter(&$form, &$form_state) {
 function orwell_form_user_login_block_alter(&$form, &$form_state) {
   // Hide links.
   $form['links']['#access'] = FALSE;
+
+  // Add validate handler so we can detect failed login.
+  $form['#validate'][] = 'orwel_form_user_login_block_validate';
+}
+
+/**
+ * Form validate callback.
+ *
+ * Makes a note if the login failed.
+ */
+function orwel_form_user_login_block_validate(&$form, &$form_state) {
+  $failed = &drupal_static('orwell_login_failed', FALSE);
+  if (!isset($form_state['uid']) || !$form_state['uid']) {
```

Comment:
If $form_state['uid'] is not set, this will give a notice as you've used OR

Shouldnt this just be 
```
if (empty($form_state['uid']))

Reply:
If it is not set, `isset($form_state['uid'])` evaluates to false, `!isset($form_state['uid'])` evaluates to true. `||` and `&&` (and `or` and `and`) are shortcutting operators, so it wont evaluate `!$form_state['uid']` as it wont change the result of the statement (it's true as `true || X` is true regardless of whether X is true or false).

That's also why `empty($var) or print "Please set var"` works as you'd suspect by reading it out loud.

Reply:
Ok.. but couldnt you still just use `if (empty($form_state['uid']))` ?

Reply:
It could. It's just a matter of taste. My code makes it more explicit that we're expecting to deal with an unset value.